lib/web_ui/lib/src/engine/view_embedder/dom_manager.dart

```diff
@@ -52,10 +52,20 @@ class DomManager {
     final DomElement semanticsHost = domDocument.createElement(DomManager.semanticsHostTagName);
     final DomElement announcementsHost = createDomElement(DomManager.announcementsHostTagName);
 
-    // Root element children.
+    // This `flt-view-id` attribute does not serve a function in the engine's
+    // operation, but it's useful for debugging, test automation, and DOM
+    // interop use-cases. It allows one to use CSS selectors to find views by
+    // their identifiers.
+    //
+    // Example:
+    //
+    //     document.querySelector('flutter-view[flt-view-id="$viewId"]')
+    rootElement.setAttribute('flt-view-id', viewId);
```

Comment:
The embedding strategy is setting most of the `flt` values in the rootElement 

https://github.com/flutter/engine/blob/3848f505cb4367b0c49691b807a3314be17210c6/lib/web_ui/lib/src/engine/embedder.dart#L57-L66

It is likely that this will eventually move there once it is made view-aware.